src/app/icd/ICDEventManager.cpp

```diff
@@ -97,7 +98,19 @@ void ICDEventManager::ICDEventHandler(const ChipDeviceEvent * event, intptr_t ar
         {
             icdManager->UpdateOperationState(ICDManager::OperationalState::ActiveMode);
         }
+        break;
+    case DeviceEventType::kICDMsgAckSyncEvent:
```

Comment:
At this point, would it make more sense to just track a count of "exchange has sent a message but has not been destroyed", complexity-wise?  Or do we not want to count exchanges which are waiting on us to do some processing and then send a message?

Reply:
Yes indeed, I was coming to the same conclusion while doing this PR. I believe there is some small use case that would keep us in active mode longer than necessary blocking for each context.

I will probably end up reworking the mechanism around the exchange context soon. But for now this PR, with the existing mechanism, do work for ICDs.

I'll make it better in the upcoming weeks. but would like to get this in for now